go/private/rules/transition.bzl

```
@@ -173,6 +195,23 @@ def _go_transition_impl(settings, attr):
         linkmode_label = filter_transition_label("@io_bazel_rules_go//go/config:linkmode")
         settings[linkmode_label] = linkmode
 
+    for setting, value in settings.items():
+        original_value_setting = _original_value_setting(setting)
+        if not original_value_setting:
+            continue
```

```suggestion
        if not original_value_setting or value == original_settings[setting]:
            continue
```

This would reduce the nesting for the logic below.